### PR TITLE
Add support to Debian 9

### PR DESCRIPTION
**Pull request by bhack**
I've not actually a Debian host with Nvidia GPU to test it.
Have you a K8 Debian 8 GPU node to test it?

### CODE REVIEW

File: cmd/nvidia_gpu/README.md [hunk 1]
```diff
@@ -18,4 +18,9 @@ Examples of how driver installation needs to be done can be found at:
   - Installer code: https://github.com/GoogleCloudPlatform/container-engine-accelerators/blob/master/nvidia-driver-installer/ubuntu/entrypoint.sh
   - Installer daemonset: https://github.com/GoogleCloudPlatform/container-engine-accelerators/blob/master/nvidia-driver-installer/ubuntu/daemonset.yaml
 
+- For [Debian](https://cloud.google.com/kubernetes-engine/docs/concepts/node-images#debian) (experimental):
```
Comment (mindprince) [inline cmd/nvidia_gpu/README.md]: This link is 404.

File: nvidia-driver-installer/debian/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        hostPath:
+          path: /dev
+      - name: nvidia-install-dir-host
+        hostPath:
+          path: /home/kubernetes/bin/nvidia
+      - name: root-mount
+        hostPath:
+          path: /
+      initContainers:
+      - image: gcr.io/google-containers/debian-nvidia-driver-installer@sha256:7ffaf40fcf6bcc5bc87501b6be295a47ce74e1f7aac914a9f3e6c6fb8dd780a4
```
Comment (mindprince) [inline nvidia-driver-installer/debian/daemonset.yaml]: This is wrong. Point it to `latest` for now.

**Comment by mindprince** [comment]
Also, can you please squash your commits?

File: cmd/nvidia_gpu/README.md (hunk 1 above)
Comment (bhack) [reply cmd/nvidia_gpu/README.md]: I cannot find a Google reference about Debian node image docs.

File: nvidia-driver-installer/debian/daemonset.yaml (hunk 2 above)
Comment (bhack) [reply nvidia-driver-installer/debian/daemonset.yaml]: So do you want a cut before `@sha`?

**Comment by bhack** [comment]
Done.

**Comment by mindprince** [comment]
Thinking more about this, since everything is the same as the ubuntu script, would it make more sense to just add notes in the README that the ubuntu script/image works for Debian as well? It would also make sure that the ubuntu and debian scripts don't go out of sync.

**Comment by bhack** [comment]
It is not the same cause the base docker image is changing. I.e. check [this line](https://github.com/GoogleCloudPlatform/container-engine-accelerators/pull/51/files#diff-573da5fd7bbb10c8ede2e8be0162ab84R43?)

**Comment by bhack** [comment]
What I  mean is that for the kernel headers the host node and the installing container os versions needs "to match". Basically for a Debian 9 node we need a Debian 9 installing container.

**Comment by bhack** [comment]
An alternative is to introduce in ubuntu build args for the `Dockerfile`, arg variable for `Makefile`, and a variable for the `daemonset.yaml`

**Comment by mindprince** [comment]
> It is not the same cause the base docker image is changing. I.e. check [this line](https://github.com/GoogleCloudPlatform/container-engine-accelerators/pull/51/files#diff-573da5fd7bbb10c8ede2e8be0162ab84R43)

> What I mean is that for the kernel headers the host node and the installing container os versions needs "to match". Basically for a Debian 9 node we need a Debian 9 installing container.

I have yet to test it, but I don't think the base docker image being Ubuntu would be a problem.
 `uname -r` returns the host kernel version, so all we need from the container is that the apt repositories it is looking at contains that kernel version. If debian is using one of the kernel versions that are part of the ubuntu apt repositories, everything would work as expected. If not, all we need to do is to update the [ubuntu Dockerfile](https://github.com/GoogleCloudPlatform/container-engine-accelerators/blob/f56351f7b2fc48e75b3c52e5f4ae101c9aa02f0c/nvidia-driver-installer/ubuntu/Dockerfile#L17) to add the apt repository which contains the debian kernel.

> An alternative is to introduce in ubuntu build args for the Dockerfile, arg variable for Makefile, and a variable for the daemonset.yaml

This maybe preferable if using the same docker image doesn't work out.

**Comment by bhack** [comment]
Generally.the kernels mismatch but let me know when you have a Debian node with Gpu to test it.

**Comment by vishh** [comment]
What's the maintenance plan? I don't think we at Google have bandwidth to maintain additional distros at the moment.

**Comment by bhack** [comment]
@vishh How do you want to solve https://github.com/kubernetes/kubernetes/issues/54011? There is still Debian in the title.

**Comment by vishh** [comment]
@bhack If you can setup a test infra for this feature, we can consider taking it. Otherwise, we may have to wait for Nvidia to release their driver installer container images

**Comment by bhack** [comment]
I don't own test infra. What do you mean? Also this [new Nvidia announce](https://news.developer.nvidia.com/kubernetes-on-nvidia-gpus-release-candidate-now-available/) is only marketing or there is something new?

**Comment by vishh** [comment]
How do you plan on testing this PR continually against different Debian
releases?
 @3XX0 might be able to share more about Nvidia's driver installation
containers. He had demo'ed it previously in a k8s WG meeting.

**Comment by bhack** [comment]
Are we talking of https://github.com/NVIDIA/nvidia-container-runtime/?

**Comment by bhack** [comment]
I see in the distro matrix at https://nvidia.github.io/nvidia-container-runtime/ that Debian is covered. So what we need to do?

**Comment by flx42** [comment]
@bhack look at this document:
https://docs.google.com/presentation/d/1NY4X2K6BMaByfnF9rMEcNq6hS3NtmOKGTfihZ44zfrw/edit?usp=sharing

**Pull request closed without merging**